src/msgpackdump.d

```diff
@@ -9,99 +9,104 @@ import std.stdio;
 import std.exception;
 import core.bitop : bswap;
 
-private ushort bswap(ushort u)
+enum hasNoUnsignedSwap = __VERSION__ < 2088;
+
+version(hasNoUnsignedSwap)
 {
-	version (LittleEndian)
+	private ushort bswap(ushort u)
 	{
-		version (D_InlineAsm_X86_64)
+		version (LittleEndian)
 		{
-			asm
+			version (D_InlineAsm_X86_64)
 			{
-				naked;
-				mov AX, DI;
-				xchg AL, AH;
-				ret;
+				asm
+				{
+					naked;
+					mov AX, DI;
+					xchg AL, AH;
+					ret;
+				}
 			}
-		}
-		else
-		{
-			union UshortBytes
+			else
 			{
-				ushort u;
-				ubyte[2] b;
-			}
+				union UshortBytes
+				{
+					ushort u;
+					ubyte[2] b;
+				}
 
-			UshortBytes src;
-			src.u = u;
-			UshortBytes dest;
-			dest.b[1] = src.b[0];
-			dest.b[0] = src.b[1];
-			return dest.u;
+				UshortBytes src;
+				src.u = u;
+				UshortBytes dest;
+				dest.b[1] = src.b[0];
+				dest.b[0] = src.b[1];
+				return dest.u;
+			}
 		}
+		else return u;
 	}
-	else return u;
-}
 
-private ulong bswap(ulong u)
-{
-	version (LittleEndian)
+	private ulong bswap(ulong u)
 	{
-		version (D_InlineAsm_X86_64)
+		version (LittleEndian)
 		{
-			asm
+			version (D_InlineAsm_X86_64)
 			{
-				naked;
-				mov RAX, RDI;
-				bswap RAX;
-				ret;
+				asm
+				{
+					naked;
+					mov RAX, RDI;
+					bswap RAX;
+					ret;
+				}
 			}
-		}
-		else
-		{
-			union UlongBytes
+			else
 			{
-				ulong u;
-				ubyte[8] b;
-			}
+				union UlongBytes
+				{
+					ulong u;
+					ubyte[8] b;
+				}
 
-			UlongBytes src;
-			src.u = u;
-			UlongBytes dst;
-			dst.b[0] = src.b[7];
-			dst.b[1] = src.b[6];
-			dst.b[2] = src.b[5];
-			dst.b[3] = src.b[4];
-			dst.b[4] = src.b[3];
-			dst.b[5] = src.b[2];
-			dst.b[6] = src.b[1];
-			dst.b[7] = src.b[0];
-			return dst.u;
+				UlongBytes src;
+				src.u = u;
+				UlongBytes dst;
+				dst.b[0] = src.b[7];
+				dst.b[1] = src.b[6];
+				dst.b[2] = src.b[5];
+				dst.b[3] = src.b[4];
+				dst.b[4] = src.b[3];
+				dst.b[5] = src.b[2];
+				dst.b[6] = src.b[1];
+				dst.b[7] = src.b[0];
+				return dst.u;
+			}
 		}
+		else return u;
 	}
-	else return u;
 }
 
 private T read16(T)(ref size_t index, const ubyte[] bytes) if (T.sizeof == 2)
 {
 	auto buf = bswap(*(cast(ushort*) (bytes.ptr + index)));
-        T r = *(cast(T*) &buf);
-        index += 2;
+		T r = *(cast(T*) &buf);
+		index += 2;
```

Comment:
These lines and the corresponding lines in `read32` and `read64` should only be indented one level.

Reply:
I see